src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -86,44 +86,44 @@ internal partial class DiagnosticAnalyzerService
             Contract.ThrowIfFalse(project.SupportsCompilation);
             AssertCompilation(project, compilation);
 
-            // in IDE, we always set concurrentAnalysis == false otherwise, we can get into thread starvation due to
-            // async being used with synchronous blocking concurrency.
-            var projectAnalyzerOptions = new CompilationWithAnalyzersOptions(
-                options: project.AnalyzerOptions,
-                onAnalyzerException: null,
-                analyzerExceptionFilter: GetAnalyzerExceptionFilter(),
-                concurrentAnalysis: false,
-                logAnalyzerExecutionTime: true,
-                reportSuppressedDiagnostics: true);
-            var hostAnalyzerOptions = new CompilationWithAnalyzersOptions(
-                options: project.HostAnalyzerOptions,
-                onAnalyzerException: null,
-                analyzerExceptionFilter: GetAnalyzerExceptionFilter(),
-                concurrentAnalysis: false,
-                logAnalyzerExecutionTime: true,
-                reportSuppressedDiagnostics: true);
-
-            // Create driver that holds onto compilation and associated analyzers
-            return new CompilationWithAnalyzersPair(
-                filteredProjectAnalyzers.Any() ? compilation.WithAnalyzers(filteredProjectAnalyzers, projectAnalyzerOptions) : null,
-                filteredHostAnalyzers.Any() ? compilation.WithAnalyzers(filteredHostAnalyzers, hostAnalyzerOptions) : null);
-
-            Func<Exception, bool> GetAnalyzerExceptionFilter()
+            var exceptionFilter = (Exception ex) =>
             {
-                return ex =>
+                if (ex is not OperationCanceledException && crashOnAnalyzerException)
```

Comment:
>crashOnAnalyzerException [](http://example.com/codeflow?start=60&length=24)

this is the bit that's preventing this from being a static local function?

Reply:
will check.